src/lib/y2network/widgets/routing_buttons.rb

```diff
@@ -0,0 +1,68 @@
+require "cwm/common_widgets"
+require "y2network/dialogs/route"
+require "y2network/route"
+
+Yast.import "Routing"
+
+module Y2Network
+  module Widgets
+    class AddRoute < CWM::PushButton
+      def initialize(table)
+        @table = table
+        textdomain "network"
+      end
+
+      def label
+        _("Ad&d")
+      end
+
+      def handle
+        route = Y2Network::Route.new
+        res = Y2Network::Dialogs::Route.run(route, Yast::Routing.GetDevices + ["-"])
+        @table.add_route(route) if res == :ok
+
+        nil
+      end
+    end
+
+    class EditRoute < CWM::PushButton
+      def initialize(table)
+        @table = table
+        textdomain "network"
+      end
+
+      def label
+        _("&Edit")
+      end
+
+      def handle
+        return nil unless @table.selected_route
+
+        route = @table.selected_route.dup
```

Comment:
It might cause problem with nested objects. However, I've implemented a method to properly clone the config objects (not ready yet).

Reply:
as I assign new objects to route it should not be problem. As I do not modify IPAddr or Interface and instead create new object. So I just switching references, not modifying nested objects. But if in future we need to modify object that is part of table, we need to do it.